Sources/VaporToolbox/New.swift

```diff
@@ -0,0 +1,291 @@
+import ArgumentParser
+import Foundation
+
+extension Vapor {
+    struct New: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Generates a new app.")
+
+        @Argument(help: "Name of project and folder.")
+        var name: String
+
+        /// Dynamic variables taken from the template manifest.
+        var variables: [String: Any] = [:]
+
+        /// Options shared by all templates, independent of the manifest.
+        ///
+        /// They control the build process of the project.
+        struct BuildOptions: ParsableArguments {
+            @Option(
+                name: [.short, .customShort("T"), .long],
```

Comment:
I think we can rip out the upper case `T` - this is a new major version so breaking changes are fine